src/components/CryptoFinder.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from "react";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+import CryptoCard from "./CryptoCard";
+import CoinGeckoResponse from "../models/CoinGeckoResponse";
+import "./CryptoFinder.css";
+
+type CryptoList = {
+  id: string;
+  symbol: string;
+  name: string;
+};
+
+type CryptoInfo = {
+  id: string;
+  name: string;
+};
+
+const fetchCryptoData = async (id: string) => {
+  const response = await axios.get<CoinGeckoResponse>(
+    `https://api.coingecko.com/api/v3/simple/price?ids=${id}&vs_currencies=usd,eur,gbp&precision=2`
+  );
+  return response.data[id];
+};
+
+const fetchCryptoList = async (): Promise<CryptoList[]> => {
+  const response = await axios.get<CryptoList[]>(
+    `https://api.coingecko.com/api/v3/coins/list`
+  );
+  return response.data;
+};
+
+const CryptoFinder: React.FC<{
+  onSelectCrypto: (crypto: CryptoInfo) => void;
+  favCryptos: CryptoInfo[];
+}> = (props) => {
+  const [selectedCoin, setSelectedCoin] = useState<CryptoInfo>({
+    id: "",
+    name: "",
+  });
+
+  const [queryCoin, setQueryCoin] = useState<CryptoInfo>({
+    id: "",
+    name: "",
+  });
+
+  const [cryptoList, setCryptoList] = useState<CryptoList[]>([]);
+
+  useEffect(() => {
+    const getCryptos = async () => {
+      const cryptos = await fetchCryptoList();
+      setCryptoList(cryptos);
```

Comment:
why is this not part of another useQuery?

Reply:
This is used to fetch the list of all available cryptos, which will change quite infrequently. Also, coingeckoapi has a very harsh rate limiter, so I wanted to reduce the number of API calls made. UseQuery will trigger the request after the data in cache becomes stale/ if a user comes back to this page from another tab.

Reply:
You can specify those parameters in useQuery as well.